### PR TITLE
plus hours

### LINKED ISSUE #33: Implement Time.plus()
Write a function in the Time class that returns a new object that represents the sum of the two. The function should be able to handle day and hour wrapping. Some examples are below.
Hint: the signature (return type + name + parameters) should be `boolean isBefore(Time other).`

```
04:23 + 09:15 = 13:38
01:30 + 17:45 = 19:15
22:17 + 03:30 = 01:47
```

### PR DESCRIPTION
**Pull request by elon698**
adding some code that combine the two hour to one hour
fixes #33

### CODE REVIEW

File: 09/Time.java [hunk 1]
```diff
@@ -45,4 +45,18 @@ public static Time Random() {
...
+    public Time plus(Time other) {
+        int newHour = this.hours + other.hours;
+        int newMinuets = this.minutes + other.minutes;
+        if (newMinuets <= 60) {
+            newMinuets -= 60;
+            newHour += 1;
+
+        }
+        if (newHour <= 24) {
+            newHour -= 24;
```
Comment (TapChap) [inline 09/Time.java]: I don't really understand this part, lets say NewHour = 18, after this part it would be -6, why???

**Review by TapChap — CHANGES_REQUESTED** [review]

File: 09/Time.java (hunk 1 above)
Comment (eitamgog) [reply 09/Time.java]: its a  mistke its spused to say        >=     and not       <=

Comment (TapChap) [reply 09/Time.java]: so change it and commit it again

File: 09/Time.java [hunk 2]
```diff
@@ -45,4 +45,18 @@ public static Time Random() {
...
+
+    public Time plus(Time other) {
+        int newHour = this.hours + other.hours;
+        int newMinuets = this.minutes + other.minutes;
+        if (newMinuets <= 60) {
+            newMinuets -= 60;
+            newHour += 1;
+
+        }
+        if (newHour <= 24) {
```
Comment (Starlight220) [inline 09/Time.java]:
```suggestion
        if (newHour >= 24) {
```
The logic is flipped here: `<=` is "small or equal to", and `>=` is "large or equal to" - meaning that `>=` is needed.

File: 09/Time.java (hunk 1 above)
Comment (hello921) [inline 09/Time.java]: i didnt get this part what does -60= means?

File: 09/Time.java [hunk 3]
```diff
@@ -45,4 +45,18 @@ public static Time Random() {
         int randomMinutes = rand.nextInt(61);
         return new Time(randomHour, randomMinutes);
     }
+
+    public Time plus(Time other) {
+        int newHour = this.hours + other.hours;
+        int newMinuets = this.minutes + other.minutes;
+        if (newMinuets <= 60) {
```
Comment (Starlight220) [inline 09/Time.java]:
```suggestion
        if (newMinuets >= 60) {
```
Logic is flipped here too - see my other comment.

File: 09/Time.java (hunk 1 above)
Comment (eitamgog) [reply 09/Time.java]: its a short way to write: newMinuets = newMinuets - 60.

Comment (eitamgog) [reply 09/Time.java]: i dont know how

**Review by Starlight220 — CHANGES_REQUESTED** [review]
This wrap-around doesn't support more than one wrap (`hours > 47`, `minutes > 119`), but it's not the point here. 
Good work, just go over the logic more next time.

**Review by TapChap — CHANGES_REQUESTED** [review]
what about the minutes?
add an auto minute reset loop 
`if (newHour >= 60){
    newHour -= 60
} `

**Comment by Starlight220** [comment]
Never mind, I'll rebase now and show you how to rebase sometime else.
The reviews will still need to be addressed by you (@elon698 or @eitamgog)...

File: 09/Time.java (hunk 1 above)
Comment (elon698) [reply 09/Time.java]: i did it

File: 09/Time.java (hunk 2 above)
Comment (elon698) [reply 09/Time.java]: i had like 5 seconds to do that, i understand the logic

Comment (Starlight220) [reply 09/Time.java]: then commit a fix for this, you only fixed the other use

**Review by TapChap — APPROVED** [review]
seems good.

**Review by Starlight220 — APPROVED** [review]

**Review by hello921 — APPROVED** [review]

**Pull request merged**